File: 백준/Bronze/23027. 1번부터 문제의 상태가…？/1번부터 문제의 상태가…？.js
const fs = require('fs');
let input = fs.readFileSync(0).toString().trim();

function change(letter) {
  if (letter.includes('A')) {
    // /[BCDF]/는 정규식. 전체에서(g) B,C,D,F를 찾는 것
    return letter.replace(/[BCDF]/g, 'A');
  } else if (letter.includes('B')) {
    return letter.replace(/[CDF]/g, 'B');
  } else if (letter.includes('C')) {
    return letter.replace(/[DF]/g, 'C');
  } else {
    return 'A'.repeat(letter.length);
  }
}

console.log(change(input));

File: 백준/Bronze/23027. 1번부터 문제의 상태가…？/1번부터 문제의 상태가…？.py
S = input().strip()

if 'A' in S:
    result = ''.join('A' if c in 'BCDF' else c for c in S)
elif 'B' in S:
    result = ''.join('B' if c in 'CDF' else c for c in S)
elif 'C' in S:
    result = ''.join('C' if c in 'DF' else c for c in S)
else:
    result = 'A' * len(S)

print(result)